ui/common/Input.tsx

```diff
@@ -1,39 +1,93 @@
-'use client'
-
-import RICIBs from 'react-individual-character-input-boxes'
-
-export default function Input({
-  amount,
-  onChange,
-}: {
-  amount: number
-  onChange: Function
-}) {
-  const handleChange = (value: string) => {
-    onChange(value)
-  }
-
-  return (
-    <RICIBs
-      amount={amount}
-      autoFocus
-      handleOutputString={handleChange}
-      inputProps={{
-        className: 'bg-transparent',
-        placeholder: '_',
-        style: {
-          fontSize: '20px',
-          width: '20px',
-          height: '20px',
-          margin: '0px',
-          borderRadius: '0px',
-          textAlign: 'center',
-          justifyContent: 'space-evenly',
-          outline: 'none',
-          fontFamily: 'var(--space-mono-font)',
-        },
-      }}
-      inputRegExp={/^[a-zA-Z0-9_.-]*$/}
-    />
-  )
-}
+'use client'
+
+import { useState } from 'react'
+
+interface UserInputProps {
+  onChange: Function
+  answer: string
+  hints?: boolean
+}
+
+export default function Input({ onChange, answer, hints }: UserInputProps) {
+  const [textAreaValue, setTextAreaValue] = useState('')
+  const [correctAns, setCorrectAns] = useState(false)
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLTextAreaElement>) => {
+    if (event.key === ' ') {
+      event.preventDefault()
+    }
+  }
+
+  const blankSpace = () => {
+    let underscores = '_'.repeat(answer.length)
+    underscores = underscores.substring(0, answer.length - textAreaValue.length)
+    return (
+      <>
+        {Array.from({ length: answer.length }, (_, i) => {
+          if (i >= answer.length - underscores.length)
+            return (
+              <span className="underscore" key={i}>
+                _
+              </span>
+            )
+          if (textAreaValue[i] === answer[i]) {
+            return (
+              <span className="overlay correct" key={i}>
+                {textAreaValue[i]}
+              </span>
+            )
+          } else if (textAreaValue[i] !== answer[i] && !!hints) {
+            return (
+              <span className="overlay incorrect" key={i}>
+                {textAreaValue[i]}
+              </span>
+            )
+          } else {
+            return (
+              <span className="overlay" key={i}>
+                {textAreaValue[i]}
+              </span>
+            )
+          }
+        })}
+      </>
+    )
+  }
+
+  return (
+    <form className="relative">
+      <textarea
+        onKeyDown={handleKeyDown}
+        onChange={(e) => {
+          let noSpace = e.target.value.replace(/[\s\u00A0]+/g, '').replace(/\r?\n|\r/g, '').slice(0, answer.length).toLowerCase()

```

Comment:
Let's try to keep the jsx markup separated from the logic, this helps with readability. So what I mean by that is simply create a handler function below `handleKeyDown`, maybe call it `handleChange` or something akin to that. Also makes things easier to find.